LiteCore/Storage/SQLiteDataFile.cc

```diff
@@ -796,7 +796,7 @@ namespace litecore {
         if (collection == "_" || slice(collection) == KeyStore::kDefaultCollectionName || slice(collection) == KeyStore::kDefaultFullCollectionName)
             name += kDefaultKeyStoreName;
         else {
-            string candidate = name + string(KeyStore::kCollectionPrefix) + collection;
+            string candidate = name + string(KeyStore::kCollectionPrefix) + SQLiteKeyStore::transformCollectionName(collection, true);
```

Comment:
Why not hide the case mangling inside tableExists, before it hits SQLite table query? The case-mangling is a quirky thing owing to SQLite.

Reply:
I think the current fix seems to appropriate to me as the function itself needs to return the table name as well.

Reply:
" tableExists() will need to return ..." -- I assume you meant "collectionTableName() will ..."
It is opposite: with this change, collectionTableName() will return mangled table name.